src/pykeen/triples/triples_factory.py

```diff
@@ -640,15 +645,18 @@ def new_with_restriction(
             relations = list(relations) + list(map(self.relation_to_inverse.__getitem__, relations))
 
         keep_mask = None
+        tags = {}
 
         # Filter for entities
         if entities is not None:
             keep_mask = self.get_idx_for_entities(entities=entities)
+            tags['entities'] = sorted(set(entities))
```

Comment:
this could get quite long

Reply:
good point. this probably isn't actually a good idea